presto-spi/src/test/java/io/prestosql/spi/type/TestChars.java

```diff
@@ -18,13 +18,39 @@
 
 import static io.airlift.slice.Slices.utf8Slice;
 import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.prestosql.spi.type.CharType.createCharType;
 import static io.prestosql.spi.type.Chars.byteCountWithoutTrailingSpace;
+import static io.prestosql.spi.type.Chars.padSpaces;
 import static io.prestosql.spi.type.Chars.truncateToLengthAndTrimSpaces;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.fail;
 
 public class TestChars
 {
+    @Test
+    public void testPadSpaces()
+    {
+        testPadSpaces("", 0, "");
+        testPadSpaces("", 1, " ");
+        testPadSpaces("", 3, "   ");
+        testPadSpaces("ab", 3, "ab ");
+        testPadSpaces("abc", 3, "abc");
+        testPadSpaces("🚁", 3, "🚁  ");
+        testPadSpaces("🚁🚁🚁", 3, "🚁🚁🚁");
```

Comment:
Don't use unicode characters in source code -- some tools may not play well with that. Replace them with the corresponding \u escape sequences.

Reply:
https://github.com/airlift/airbase/blob/master/airbase/pom.xml#L81

https://github.com/prestosql/presto/blob/b669b5762592d654393a52a792410ec2dd545ffd/presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java#L182-L183

Do you have particular tools in mind?

Reply:
Nothing concrete at this point, but I can think of a bunch of things that could fail to display those characters (Unix terminals that don't support or are not configured to display UTF-8, etc).

Reply:
This is what default emacs on MacOS shows, for instance:

![Screen Shot 2019-07-23 at 4 12 53 PM](https://user-images.githubusercontent.com/14387/61753666-09d88e00-ad65-11e9-8f2e-b6a180653d11.png)

Reply:
> https://github.com/airlift/airbase/blob/master/airbase/pom.xml#L81

That's only for maven :)

Reply:
My concern is that unicode escapes make the code less readable and unicode escapes with surrogate pairs even more so...
In this particular case there is one advantage of using unicode escapes and I will change the code, but I don't agree in general.
Unless you convince me the tools than cannot be fixed to handle UTF-8 are important to us.

Reply:
![image](https://user-images.githubusercontent.com/144328/61771057-186b8780-adef-11e9-8bcc-e642a182bd9b.png)

😄

Reply:
I decided not to change `🚁` to `\ud83d\ude81`. With the latter, it's not at all visible that this is a surrogate pair and not just some 2 characters. 
Instead, I introduced a variable conveying this explicitly:

```java
String nonBmp = "🚁";
verify(nonBmp.codePoints().count() == 1, "nonBmp must be a single character (code point) string");
verify(nonBmp.length() == 2, "nonBmp must be encoded as surrogate pairs (2 Java characters)");
```

and I used this variable in following test code, instead of literal.


(I am still happy to talk about whether source code should be UTF-8, or UTF-8 sans non-BMP, or ASCII. This PR doesn't change anything with this regard, so let's move this offline.)

Reply:
Given that the first char of a surrogate pair has to be in the D8xx-DFxx, does that make it fairly obvious it’s a surrogate pair? (More than the helicopter icon)

Reply:
> Given that the first char of a surrogate pair has to be in the D8xx-DFxx, 

I failed to intern this knowledge.

> More than the helicopter icon

I think all emojis are surrogate pairs in UTF-16.

Reply:
Whatever, `\uD83D\uDE81"` it will be.